libraries/dagster-contrib-weaviate/dagster_contrib_weaviate/resource.py

```diff
@@ -0,0 +1,190 @@
+            " Weaviate instance. You may provide any of the following options:"
+            " api_key / access_token (bearer_token) / username+password / client_secret"
+            " See https://weaviate.io/developers/weaviate/configuration/authentication for more info."
+        ),
+        default={},
+    )
+
+    def _weaviate_auth_credentials(self) -> Optional[weaviate.auth.AuthCredentials]:
+        """Converts the auth_credentials config dict from the user, to the Weaviate AuthCredentials
+        class that can be passed to the WeaviateClient constructor."""
+
+        if len(self.auth_credentials) == 0:
+            return None
+
+        if "api_key" in self.auth_credentials:
+            return weaviate.classes.init.Auth.api_key(**self.auth_credentials)
+
+        if "access_token" in self.auth_credentials:
+            return weaviate.classes.init.Auth.bearer_token(**self.auth_credentials)
+
+        if "username" in self.auth_credentials:
+            return weaviate.classes.init.Auth.client_password(**self.auth_credentials)
+
+        if "client_secret" in self.client_credentials:
+            return weaviate.classes.init.Auth.client_credentials(
+                **self.auth_credentials
+            )
+
+        raise Exception(
+            "One of the following must be provided in auth_credentials configuration:"
+            " api_key / access_token / username+password / client_secret"
+        )
+
+
+class WeaviateCloudResource(WeaviateBaseResource):
+    """Resource for interacting with a Weaviate Cloud database.
+
+    Use WeaviateLocalResource to interact with a self hosted Weaviate resource.
+
+    Examples:
+        .. code-block:: python
+
+            from dagster import Definitions, asset
+            from dagster_contrib_weaviate import WeaviateCloudResource
+
+            @asset
+            def my_table(weaviate: WeaviateCloudResource):
+                with weaviate.get_client() as weaviate_client:
+                    questions = weaviate_client.collections.get("Question")
+                    questions.query.near_text(query="biology", limit=2)
+
+            defs = Definitions(
+                assets=[my_table],
+                resources={
+                    "weaviate": WeaviateCloudResource(
+                        cluster_url=wcd_url,
+                        auth_credentials={
+                            "api_key": wcd_apikey
+                        },
+                        headers={
+                            "X-Cohere-Api-Key": cohere_apikey,
+                        }
+                    ),
+                },
+            )
+
+    """
+
+    cluster_url: str = Field(
+        description=(
+            "The WCD cluster URL or hostname to connect to. Usually in the form: rAnD0mD1g1t5.something.weaviate.cloud"
+        )
+    )
+
+    @classmethod
+    def _is_dagster_maintained(cls) -> bool:
+        return False
+
+    @contextmanager
+    def get_client(self):
```

Comment:
Could we add a return type here? I think it should be `Generator[WhateverClientTypeIs, None, None]`